IanGRap/Squish
# Refactor lobby canvas elements

Contributors: Alfredo Cuevas

Goal: Create a section in the hierarchy for Canvas Objects. Refactor some canvas game objects

Changes: 
* Found all game objects in the Lobby that had canvases in them. Moved them into the new canvas section. 
* Its generally **not** good practice to child 3D objects inside canvases. Also it's preferred to separate 3D models from the canvases and have them be children to the same empty GameObject.
  * I tried to refactor most of the items to  abide by the above rules.
* Like before I tried to avoid code changes in this pull request. The only script that I did change was LobbyArrowBounce.cs. This script now changes the position of the arrow using RectTransform. This is supposed to be the better for UI than just using transform.
* I recreated the Exit Sign object because before it was set up with values that were difficult to work with
* Added Event Cameras to all of the world space canvases because this is slightly better for performance then leaving it empty
* Minor changes to the scene happened in the process of refactoring game objects

How to test: 
* The game should run the same as before. 
* Play through one game. Focus on the Lobby. Make sure nothing is broken.
* Look at the hierarchy. Does the new Canvas section look alright. Does the hierarchy look better
* Look at the lobby. I moved a few minor things in the scene. Make sure that the lobby still looks fine. If you don't notice a difference then that is ideal. 

Notes: 
* I originally wanted to make sure that all canvas objects followed the same structure and were prefabbed. However, I ran into a lot of issues with some of the game objects (which is why this pull request took longer than it should have). The main issue is that the code heavily relies on a certain hierarchical structure. So I won't be able to adjust some of these canvas objects until I refactor some of the UI code. This will be done in the coming lobby refactor.
